packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,286 @@
+import { toRefObject, Ref } from '@stardust-ui/react-component-ref'
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import * as PropTypes from 'prop-types'
+import * as _ from 'lodash'
+
+import {
+  applyAccessibilityKeyHandlers,
+  childrenExist,
+  AutoControlledComponent,
+  RenderResultConfig,
+  isBrowser,
+  ChildrenComponentProps,
+  ContentComponentProps,
+  StyledComponentProps,
+  commonPropTypes,
+  isFromKeyboard,
+  setWhatInputSource,
+} from '../../lib'
+import { ShorthandValue, Props } from '../../types'
+import {
+  ALIGNMENTS,
+  POSITIONS,
+  Popper,
+  BasicPositioningProps,
+  PopperChildrenProps,
+} from '../../lib/positioner'
+import TooltipContent from './TooltipContent'
+import { Accessibility } from '../../lib/accessibility/types'
+import { ReactAccessibilityBehavior } from '../../lib/accessibility/reactTypes'
+
+export interface TooltipSlotClassNames {
+  content: string
+}
+
+export interface TooltipState {
+  open: boolean
+}
+
+export interface TooltipProps
+  extends StyledComponentProps<TooltipProps>,
+    ChildrenComponentProps,
+    ContentComponentProps<ShorthandValue>,
+    BasicPositioningProps {
+  /**
+   * Accessibility behavior if overridden by the user.
+   * */
+  accessibility?: Accessibility
+
+  /** Additional CSS class name(s) to apply.  */
+  className?: string
+
+  /** Existing element the tooltip should be bound to. */
+  mountNode?: HTMLElement
+
+  /** Delay in ms for the mouse leave event, before the tooltip will be closed. */
+  mouseLeaveDelay?: number
+
+  /** A tooltip can show a pointer to trigger. */
+  pointing?: boolean
+
+  /**
+   * DOM element that should be used as tooltip's target - instead of 'trigger' element that is used by default.
+   */
+  target?: HTMLElement
+
+  /** Element to be rendered in-place where the tooltip is defined. */
+  trigger?: JSX.Element
+}
+
+/**
+ * A tooltip that displays information related to an element when the element receives keyboard focus
+ * or the mouse hovers over it.
+ */
+export default class Tooltip extends AutoControlledComponent<TooltipProps, TooltipState> {
+  static displayName = 'Tooltip'
+
+  static className = 'ui-tooltip'
+
+  static slotClassNames: TooltipSlotClassNames = {
+    content: `${Tooltip.className}__content`,
+  }
+
+  static Content = TooltipContent
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      animated: false,
+      as: false,
+      content: false,
+    }),
+    align: PropTypes.oneOf(ALIGNMENTS),
+    defaultOpen: PropTypes.bool,
+    inline: PropTypes.bool,
+    mountNode: customPropTypes.domNode,
+    mouseLeaveDelay: PropTypes.number,
+    open: PropTypes.bool,
+    onOpenChange: PropTypes.func,
+    pointing: PropTypes.bool,
+    position: PropTypes.oneOf(POSITIONS),
+    target: PropTypes.any,
```

Comment:
```suggestion
    target: customPropTypes.domNode,
```

Should be more accurate